nucypher/utilities/prometheus/collector.py

```diff
@@ -305,10 +305,36 @@ def _get_arg_metric_key(self, arg_name: str):
         return f'{self.event_name}_{arg_name}'
 
 
+class CommitmentMadeEventMetricsCollector(EventMetricsCollector):
+    """Collector for CommitmentMade event."""
+    def __init__(self, staker_address: ChecksumAddress, event_name: str = 'CommitmentMade', *args, **kwargs):
+        super().__init__(event_name=event_name, argument_filters={'staker': staker_address}, *args, **kwargs)
+        self.staker_address = staker_address
+
+    def initialize(self, metrics_prefix: str, registry: CollectorRegistry) -> None:
+        super().initialize(metrics_prefix=metrics_prefix, registry=registry)
+
+        missing_commitments = self.contract_agent.get_missing_commitments(checksum_address=self.staker_address)
+        if missing_commitments == 0:
+            # has either already committed to this period or the next period
+
+            # use local event filter for initial data
+            last_committed_period = self.contract_agent.get_last_committed_period(staker_address=self.staker_address)
+            arg_filters = {'staker': self.staker_address, 'period': last_committed_period}
+            # check from the beginning - difficult to determine blocknumber for previous period
+            # however, 'staker' and 'period' filters should help reduce load
+            initial_event_filter = self.contract_agent.contract.events[self.event_name].createFilter(
+                fromBlock=0, argument_filters=arg_filters)
+            events = initial_event_filter.get_all_entries()
+            # we expect only one event to be returned due to period filter
+            for event in events:
+                self._event_occurred(event)
```

Comment:
Method looks good. But it possible to have multiple events for one period. Not sure is it an issue or not

Reply:
✔️